.github/workflows/main.yml

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Daffodil Schema Template CI
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    name: Java ${{ matrix.java_version }}, Scala ${{ matrix.scala_version }}, ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        java_version: [ 8, 11, 15 ]
+        scala_version: [ 2.12.11 ]
+        os: [ 'ubuntu-latest' ]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+
+      ############################################################
+      # Setup
+      ############################################################
+
+      - name: Checkout Repository
+        uses: actions/checkout@v2.0.0
```

Comment:
Why `@v2.0.0` instead of `@v2`?  This prevents the workflow from using any later version in the v2 series.

Reply:
I know we've ran into an issue at one point where github made an update to the action that had the same major version and it actually broke something. To avoid that, we changed our actions to specify a specific version. I'm fine updating to a newer version if one exist, but to prevent random breakages, I sort of like to stick with known good versions.

Reply:
Open ended version dependencies have been used for supply-chain security attacks recently. The only secure practice is to specify exact version numbers of all dependencies. The hack is called "Dependency Hijacking"